config.default.toml

```diff
@@ -0,0 +1,15 @@
+[fossil]
+port = 8000
+prom-port = 2112
+
+host = "fossil://local/default"
+local = true
+verbose = 2
+
+[database]
+path = "data/default.log"
```

Comment:
The way I've been thinking about a database is that it's actually the directory; not any particular file. This is because there are actually 2 files that together make up the fossil database:

```
λ › ls -al                                                                                                                                                              fossil/test2 main
total 1820
drwxr-xr-x. 1 dburkart dburkart     30 Nov 11 13:38 .
drwxr-xr-x. 1 dburkart dburkart    252 Nov 30 14:51 ..
-rw-------. 1 dburkart dburkart 990282 Nov 11 13:38 database
-rw-------. 1 dburkart dburkart 869304 Nov 11 13:38 wal.log
```

In the above example, I've created a local database at the location "./test2", and filled it up with 10's of thousands of pieces of data. You can see there are two files:

  * `database`: This is an on-disk checkpoint of our in-memory database structure. Serializing it out is very expensive, so we only do so once we meet a threshold of data.
  * `wal.log`: This is our write-ahead-log. When initializing the database, we first load up the `database` file (if it exists yet for the DB), and then replay all lines in this file to bring the snapshot up-to-date. This is necessary both for consistency (ensuring that data we add to the DB between checkpoints is likely to be recovered in a crash), and performance (it's much less expensive to update this log than serialize out the whole DB)

In the future, we'll probably also support other bits of data we'd want to store in the database directory-- or even split up the monolithic database snapshot. For example, we may want the ability to "evict" old segments to free up memory.

Reply:
Ok might make sense to just update this key to `directory`?